text/0093-nft-standard.md

```diff
@@ -0,0 +1,260 @@
+
+- Physical assets — real estate, unique artwork 
+- Digital collectibles — rare digital items, game assets, virtual companions 
+- Obligations and non-traditional assets — debt instruments, legal contracts, responsibilities
+
+Each NFT is uniquely identifiable, and as such, must be tracked individually. Unlike fungible tokens, NFTs are not interchangeable — every item is distinct, and this standard ensures that uniqueness and ownership are verifiable and transferable on-chain.
+
+## Motivation
+
+[motivation]: #motivation
+A standard interface for non-fungible tokens (NFTs) on the Casper network enables wallets, marketplaces, decentralized applications, and other ecosystem tools to interact seamlessly with any compliant NFT contract. The lessons learned from Ethereum’s ERC-721 standard show that a consistent interface increases interoperability and speeds up adoption. On Casper, this standard will ensure that every NFT, whether tracking digital collectibles or physical assets, is uniquely identifiable and transferable.
+
+## Objective
+
+[objective]: #objective
+
+The objective of this proposal is to define a Casper-compatible NFT standard that supports:
+
+- Tracking ownership and metadata of NFTs. 
+- Transferring NFTs safely between accounts. 
+- Managing per-token approvals as well as operator-level approvals.
+- (Optionally) exposing metadata (name, symbol, token URI) so that applications can easily display NFT details.
+
+This proposal also addresses the backend indexing and tracking of NFT events (minting, transfer, approvals) for full visibility into asset histories.
+
+## Core Casper NFT Interface
+
+[core-casper-nft-interface]: #core-casper-nft-interface
+
+The core interface is designed to address the basic functionality required to manage non-fungible tokens. It uses Casper’s types (e.g., AccountHash, U256) and aims to capture the semantics of ERC-721, adjusted for Casper’s execution model.
+
+```rust
+#![no_std]
+
+use casper_contract::contract_api::{runtime, storage};
+use casper_types::{Key, URef, U256, AccountHash, ContractHash};
+
+/// Enum defining events that should be emitted by NFT operations
+#[derive(Debug)]
+pub enum CEP95Event {
+    /// Emitted when an NFT is transferred, minted (from=None), or burned (to=None)
+    Transfer { from: Option<AccountHash>, to: Option<AccountHash>, token_id: U256 },
+
+    /// Emitted when a specific NFT is approved to an account
+    Approval { owner: AccountHash, approved: Option<AccountHash>, token_id: U256 },
+
+    /// Emitted when an operator is approved/disapproved for all NFTs of an owner
+    ApprovalForAll { owner: AccountHash, operator: AccountHash, approved: bool },
+}
+
+/// Casper-compatible NFT interface
+pub trait CEP95 {
+    /// Returns the number of NFTs owned by a given account
+    ///
+    /// @param owner - The account to query.
+    /// @return U256 - The number of NFTs owned.
+    fn balance_of(&self, owner: AccountHash) -> U256;
```

Comment:
Basically, in all methods `AccountHash` should be replaced with `Key` since tokens can be owned also by a contract.